File: fraud-detection-service/app/schemas/payment_event.py
from pydantic import BaseModel
from typing import Optional

class PaymentEvent(BaseModel):
    paymentIntentId: str
    userId: str
    amount: float
    currency: str
    paymentMethod: str
    merchant: str
    ipAddress: Optional[str] = None 
    deviceId: Optional[str] = None
    timestamp: int

File: payment-service/src/main/java/com/payment/dto/PaymentEvent.java
package com.payment.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PaymentEvent {

    private String paymentIntentId;

    private String userId;

    private Double amount;

    private String currency;

    private String paymentMethod;

    private String merchant;

    private String ipAddress;

    private String deviceId;

    private Long timestamp;
}


